test/utils.lua

```diff
@@ -0,0 +1,75 @@
+-- This module contains auxiliary functions for integration tests that use lua code.
+-- To use this module, you need to copy this file to the temp pytest directory for the duration
+-- of the integration tests, then use the lua require function in your instance cfg file.
+-- The copying of this file to pytest temdir is already implemented in function
+-- create_test_instance() of test/utils.py module.
+-- Just use require('utils').bind_free_port(arg[0]) inside your cfg instance file.
+
+local box = require('box')
+local socket = require('socket')
+local ffi = require('ffi')
+
+local testutils = {}
+
+function testutils.get_bound_address_old_tarantool()
+    -- Get bound via box.cfg({listen = '127.0.0.1:0'}) socket addr for tarantool major version 1.
+    -- For tarantool major version higher then 1, this function can be replaced by box.info.listen.
+    local res = {}
+    for fd = 0, 65535 do
+        local addrinfo = socket.internal.name(fd)
+        -- Assume that the socket listens on 127.0.0.1.
+        local is_matched = addrinfo ~= nil and addrinfo.host == '127.0.0.1' and
+            addrinfo.family == 'AF_INET' and addrinfo.type == 'SOCK_STREAM' and
+            addrinfo.protocol == 'tcp' and type(addrinfo.port) == 'number'
+        if is_matched then
+            addrinfo.fd = fd
+            table.insert(res, addrinfo)
+        end
+    end
+    if #res == 0 then
+        error('no TCP sockets detected')
+    end
+
+    for key, address in pairs(res) do
+        -- We need only listening, not client sockets.
+        local value = ffi.new('int[1]')
+        local len = ffi.new('size_t[1]', ffi.sizeof('int'))
+        -- Filter socket with SO_REUSEADDR.
+        local reuseaddr = ffi.C.getsockopt(
+            address.fd,
+            socket.internal.SOL_SOCKET,
+            socket.internal.SO_OPT[socket.internal.SOL_SOCKET].SO_REUSEADDR.iname,
+            value, len
+        )
+        if reuseaddr == 0 then
+            return ('%s:%s'):format(res[key].host, res[key].port)
+        end
```

Comment:
It is possible that the process has several SO_REUSEADDR sockets. I think it is good to detect this situation and raise an error expicitly rather than choose an arbirtary address.

Reply:
Done

Reply:
Not precisely. I followed this discussion [here](https://github.com/tarantool/tt/pull/112#discussion_r989257844).